src/contractTest/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/HomeOfficeConsumerTest.java

```diff
@@ -72,7 +163,89 @@ public void should_get_appeal_details_with_ho_reference(MockServer mockServer) t
                 .relaxedHTTPSValidation()
                 .contentType(MediaType.APPLICATION_JSON_VALUE)
                 .when()
-                .post(mockServer.getUrl() + HOMEOFFICE_API_URL)
+                .post(mockServer.getUrl() + HOMEOFFICE_API_SEARCH_URL)
+                .then()
+                .statusCode(200)
+                .and()
+                .extract()
+                .body()
+                .asString();
+
+        JSONObject response = new JSONObject(actualResponseBody);
+        JSONArray status = response.getJSONArray("status");
+
+        assertThat(actualResponseBody).isNotNull();
+        assertThat(response).hasNoNullFieldsOrProperties();
+        assertThat(response.getJSONObject("messageHeader")).isNotNull();
+        assertThat(response.getJSONObject("messageHeader").getString("correlationId")).isNotBlank();
+        assertThat(response.getJSONObject("messageHeader").getString("eventDateTime")).isNotBlank();
+        assertThat(response.getJSONObject("messageHeader").getJSONObject("consumer")).isNotNull();
+        assertThat(response.getJSONObject("messageHeader").getJSONObject("consumer").getString("code"))
+            .isNotBlank();
+        assertThat(response.getJSONObject("messageHeader").getJSONObject("consumer").getString("description"))
+            .isNotBlank();
+        assertThat(response.getString("messageType")).isNotBlank();
```

Comment:
Could you check that the messageType is one of the valid Enum values here?  Would be a better check.  Same for any other Enums.

Reply:
I have added StringMatcher in the contract and check in assert